src/Aquifer.Jobs/Subscribers/TranslationSubscriber.cs

```diff
@@ -0,0 +1,379 @@
+                translatedContentHtmlItems.Add(translatedContentHtmlItem);
+
+                wordCount += HtmlUtilities.GetWordCount(translatedContentHtmlItem);
+
+                // each step gets its own machine translation record
+                _dbContext.ResourceContentVersionMachineTranslations
+                    .Add(new ResourceContentVersionMachineTranslationEntity
+                    {
+                        ResourceContentVersionId = resourceContentVersion.Id,
+                        DisplayName = translatedDisplayName,
+                        Content = translatedContentHtmlItem,
+                        ContentIndex = contentIndex,
+                        UserId = null,
+                        SourceId = MachineTranslationSourceId.OpenAi,
+                        RetranslationReason = null,
+                    });
+            }
+
+            translatedContentJson = TiptapConverter.ConvertHtmlItemsToJson(translatedContentHtmlItems);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(
+                ex,
+                "An error occurred during translation. Resource Content ID: {ResourceContentId}; Resource Content Version ID: {ResourceContentVersionId}.",
+                resourceContentId,
+                resourceContentVersion.Id);
+            throw;
+        }
+
+        resourceContentVersion.DisplayName = translatedDisplayName;
+        resourceContentVersion.Content = translatedContentJson;
+        resourceContentVersion.ContentSize = Encoding.UTF8.GetByteCount(translatedContentJson);
+        resourceContentVersion.WordCount = wordCount;
+
+        // save a snapshot of the translation
+        await _resourceHistoryService.AddSnapshotHistoryAsync(
+            resourceContentVersion,
+            oldUserId: startedByUserId,
+            oldStatus: ResourceContentStatus.TranslationAwaitingAiDraft,
+            ct);
+
+        // move to completed status
+        resourceContentVersion.ResourceContent.Status = ResourceContentStatus.TranslationAiDraftComplete;
+        await _resourceHistoryService.AddStatusHistoryAsync(
+            resourceContentVersion,
+            ResourceContentStatus.TranslationAiDraftComplete,
+            changedByUserId: startedByUserId,
+            ct
+        );
+
+        // if a Community Reviewer requested the translation then the status needs to be updated (again) now that translation has completed
+        if (translationOrigin == TranslationOrigin.CommunityReviewer)
+        {
+            resourceContentVersion.ResourceContent.Status = ResourceContentStatus.TranslationEditorReview;
+            await _resourceHistoryService.AddStatusHistoryAsync(
+                resourceContentVersion,
+                ResourceContentStatus.TranslationEditorReview,
+                changedByUserId: startedByUserId,
+                ct);
+        }
```

Comment:
I don't think there's any value in running this twice. I would just have this be an if/else.

Reply:
I thought about that but I think there's value in always having the status move from `TranslationAwaitingAiDraft` to `TranslationAiDraftComplete` because that shows that this tooling did what it was supposed to do and always does it consistently across the different translation origins.  Any objection to leaving it like this?